osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetActualPowerQualityCommandExecutorTest.java

```diff
@@ -72,14 +73,36 @@ public void before() throws ProtocolAdapterException, IOException {
 
   @Test
   void testRetrievalPublic() throws ProtocolAdapterException {
+    when(this.dlmsDevice.isPolyphase()).thenReturn(true);
     this.executeAndAssert("PUBLIC", GetActualPowerQualityCommandExecutor.getMetadatasPublic());
   }
 
   @Test
   void testRetrievalPrivate() throws ProtocolAdapterException {
+    when(this.dlmsDevice.isPolyphase()).thenReturn(true);
     this.executeAndAssert("PRIVATE", GetActualPowerQualityCommandExecutor.getMetadatasPrivate());
   }
 
+  @Test
+  void testRetrievalPublicSinglePhase() throws ProtocolAdapterException {
+    when(this.dlmsDevice.isPolyphase()).thenReturn(false);
+    this.executeAndAssert(
+        "PUBLIC",
+        GetActualPowerQualityCommandExecutor.getMetadatasPublic().stream()
+            .filter(PowerQualityObjectMetadata::isExistInSinglephase)
```

Comment:
Is this a correct test? If I change an object in the class under test from singePhase true to false, the test will still pass.

Reply:
Yes, it will test against the same list as the command executor uses, but if you change the polyphase attribute to true it will fail